sdk/src/wallet/operations/transaction/input_selection.rs

```diff
@@ -72,173 +82,76 @@ where
-                            *output_id,
-                            self.client()
-                                .get_output_mana_rewards(output_id, slot_index)
-                                .await?
-                                .rewards,
-                        );
-                    }
+            if let Some(input) = wallet_data.outputs.get(output_id) {
+                if input.output.can_claim_rewards(outputs.iter().find(|o| {
+                    input
+                        .output
+                        .chain_id()
+                        .map(|chain_id| chain_id.or_from_output_id(output_id))
+                        == o.chain_id()
+                })) {
+                    mana_rewards.insert(
+                        *output_id,
+                        self.client()
+                            .get_output_mana_rewards(output_id, creation_slot_index)
+                            .await?
+                            .rewards,
+                    );
                 }
             }
-
-            let mut input_selection = InputSelection::new(
-                available_outputs_signing_data,
-                outputs,
-                Some(wallet_data.address.clone().into_inner()),
-                slot_index,
-                protocol_parameters.clone(),
-            )
-            .with_required_inputs(mandatory_inputs)
-            .with_forbidden_inputs(forbidden_inputs)
-            .with_mana_rewards(mana_rewards);
-
-            if let Some(address) = remainder_address {
-                input_selection = input_selection.with_remainder_address(address);
-            }
-
-            if let Some(burn) = burn {
-                input_selection = input_selection.with_burn(burn.clone());
-            }
-
-            if let Some(mana_allotments) = mana_allotments {
-                input_selection = input_selection.with_mana_allotments(mana_allotments.iter());
-            }
-
-            let selected_transaction_data = input_selection.select()?;
-
-            // lock outputs so they don't get used by another transaction
-            for output in &selected_transaction_data.inputs {
-                wallet_data.locked_outputs.insert(*output.output_id());
-            }
-
-            // lock outputs so they don't get used by another transaction
-            for output in &selected_transaction_data.inputs {
-                wallet_data.locked_outputs.insert(*output.output_id());
-            }
-
-            return Ok(selected_transaction_data);
         }
 
         let mut input_selection = InputSelection::new(
             available_outputs_signing_data,
+            options.context_inputs,
             outputs,
             Some(wallet_data.address.clone().into_inner()),
-            slot_index,
+            creation_slot_index,
+            slot_commitment_id,
             protocol_parameters.clone(),
         )
+        .with_required_inputs(options.mandatory_inputs)
```

Comment:
Can we rename `mandatory_inputs` in the options to `required_inputs` for consistency? I do prefer required over mandatory.

Reply:
sure